modules/media_event_dispatcher/tests/src/Unit/Media/OEmbedResourceUrlAlterEventTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\Tests\media_event_dispatcher\Unit\Media;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\media\OEmbed\Provider;
+use Drupal\hook_event_dispatcher\HookEventDispatcherInterface;
+use Drupal\media_event_dispatcher\Event\Media\OEmbedResourceUrlAlterEvent;
+use Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Class OEmbedResourceUrlAlterEventTest.
+ *
+ * @package Drupal\Tests\media_event_dispatcher\Unit\Media
+ *
+ * @group hook_event_dispatcher
+ */
+class OEmbedResourceUrlAlterEventTest extends UnitTestCase {
+
+  /**
+   * The manager.
+   *
+   * @var \Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy
+   */
+  private $manager;
+
+  /**
+   * Sets up the test.
+   */
+  public function setUp() {
+    $builder = new ContainerBuilder();
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    \Drupal::setContainer($builder);
+  }
+
+  /**
+   * Get a mock oEmbed provider.
+   *
+   * @return \Drupal\media\OEmbed\Provider
+   *   A mock Vimeo oEmbed provider.
+   */
+  protected function getMockProvider() {
+    return new Provider('Vimeo', 'https://vimeo.com/', [
+      [
+        'url'     => 'https://vimeo.com/api/oembed.json',
+        'schemes' => [
+          'https://vimeo.com/*',
+        ],
+      ],
+    ]);
+  }
+
+  /**
+   * OEmbedResourceUrlAlterEvent parsed URL alter test.
+   *
+   * This tests altering the parsed URL array for an oEmbed data request.
+   */
+  public function testOembedResourceUrlAlterEventParsedUrlAlter() {
+    // A mock URL array as would be provided in the real hook.
```

Comment:
I have a habit of over commenting my code sometimes. I like to think it could potentially help others who are trying to learn certain things for the first time, as I sometimes had trouble figuring out the what and why of certain things in Drupal core and various contributed modules, and some of the documentation is still unclear on some things. I can definitely remove a bunch of it, but I'd recommend keeping some in for that reason.

Reply:
Yeah, I can live with that 👍. I prefer to comment only in exceptions as they tend to get outdated rather quickly. Outdated comments only spread lies 😉.

Reply:
That can definitely be an issue, especially if it's describing code elsewhere. I try to keep the documentation with the relevant code, so the likelihood of stuff getting out of date is lower that way.